src/components/ha-form/ha-form.ts

```diff
@@ -152,6 +152,10 @@ export class HaForm extends LitElement implements HaFormElement {
     `;
   }
 
+  protected fieldElementName(type: string) {
+    return `ha-form-${type}`;
+  }
```

Comment:
This should be handled in core eventually, it can provide a different type for the auth text fields.

Reply:
I thought that too at the beginning.
Don't you think we can keep this in front-end as it is a front-end issue. We can remove this code once shadow dom support is added to password manager (never? 😂)